tests/load/pipeline/test_model_item_format.py

```diff
@@ -0,0 +1,338 @@
+# test the sql insert job loader, works only on duckdb for now
+
+from typing import Any
+import pytest
+import dlt
+
+from tests.pipeline.utils import load_table_counts
+from dlt.extract.hints import make_hints, SqlModel
+from tests.load.utils import count_job_types, destinations_configs, DestinationTestConfiguration
+from dlt.common.schema.typing import TWriteDisposition
+
+from dlt.pipeline.exceptions import PipelineStepFailed
+
+DESTINATIONS_SUPPORTING_MODEL = [
+    "duckdb",
+    # "athena", #TODO: might support with the iceberg table format
+    "bigquery",
+    "clickhouse",
+    "databricks",
+    "motherduck",
+    "redshift",
+    "snowflake",
+    "sqlalchemy",
+    "mssql",
+    "postgres",
+    # "dremio",
+]
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(
+        default_sql_configs=True,
+        exclude=["athena", "dremio"],
+        subset=[
+            "snowflake",
+            "duckdb",
+            "bigquery",
+        ],
+    ),
+    ids=lambda x: x.name,
+)
+@pytest.mark.parametrize(
+    "other_destination_config",
+    destinations_configs(
+        default_sql_configs=True,
+        exclude=["athena", "dremio"],
+    ),
+    ids=lambda x: x.name,
+)
+def test_different_dialects(
+    destination_config: DestinationTestConfiguration,
+    other_destination_config: DestinationTestConfiguration,
+) -> None:
+    # NOTE: skipping these combos, something wrong with connection strings
+    if (
+        other_destination_config.destination_type == "sqlalchemy"
+        and destination_config.destination_type in ["snowflake", "bigquery"]
+    ):
+        pytest.skip("Temporarily skipping")
+    other_pipeline = other_destination_config.setup_pipeline(
+        pipeline_name="other_pipeline", dataset_name="test_model_item_format", dev_mode=False
+    )
+    other_pipeline.run(
+        [{"a": i} for i in range(10)], table_name="example_table", write_disposition="replace"
+    )
+
+    other_dataset = other_pipeline.dataset()
+    other_dialect = other_pipeline.destination.capabilities().sqlglot_dialect
+
+    example_table_columns = other_dataset.schema.tables["example_table"]["columns"]
+
+    # for now, we need to make sure the same data exists in the destination
+    pipeline = destination_config.setup_pipeline(
+        pipeline_name="pipeline", dataset_name="test_model_item_format", dev_mode=False
+    )
+    pipeline.run(
```

Comment:
please set up a pipeline with `dlt.sources.incremental`. IMO we are missing incremental `transform` for the `Model`. (it is the same mechanism we used to add extractor and storage above)

Reply:
@rudolfix what exactly is the task here? Incremental will not work with model jobs, should we have a good error or why should there be an incremental test?